packages/in_app_purchase/in_app_purchase_storekit/ios/Classes/FIAPaymentQueueHandler.m

```diff
@@ -98,34 +98,39 @@ - (instancetype)initWithQueue:(nonnull SKPaymentQueue *)queue
 }
 
 - (void)startObservingPaymentQueue {
-  self.isObservingTransactions = true;
+  self.observingTransactions = YES;
 
   [self processCachedTransactions];
 }
 
 - (void)stopObservingPaymentQueue {
-  self.isObservingTransactions = false;
+  // Don't remove the transaction observer from the SKPaymentQueue instance as
+  // that will result in transactions being lost during the current application
+  // lifetime. Only setting the "observingTransactions" to "NO" ensures
+  // incoming transactions from the App Store are cached and delivered to the
+  // client as soon as it indicates it is ready to receive transactions by
+  // sending the "startObservingPaymentQueue" message.
```

Comment:
This doesn't quite address what I was asking. My concern here is that if the Dart side stops observing, and doesn't start again, then there will be transactions in the cache that the App Store will believe have been successfully received by the application, but that will never actually get delivered anywhere. They will be thrown away, unhandled, on app exit.

Is that safe? If so, why? (E.g., will they get re-delivered the next time the app is run if they aren't acknowledged in some way?) If we don't have a guarantee from the API that this is a safe thing to do, it seems like would would need to have a persistent cache instead (e.g., using `NSUserDefaults`).

Reply:
Rephrased the comment to explain what happens when the app is killed and why it is not a problem that cached information is lost.